pkg/agent/clean/node.go

```diff
@@ -0,0 +1,349 @@
+// +build !windows
+
+package clean
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-iptables/iptables"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	"github.com/rancher/rke/hosts"
+	"github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+)
+
+const ContainerName = "cattle-node-cleanup"
+
+func Job() error {
+	logrus.Infof("Starting clean container job: %s", ContainerName)
+
+	c, err := client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	ctx := context.Background()
+
+	containerList, err := c.ContainerList(ctx, types.ContainerListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, c := range containerList {
+		for _, n := range c.Names {
+			if n == "/"+ContainerName {
+				logrus.Infof("container named %s already exists, exiting.", ContainerName)
+				return nil
+			}
+		}
+	}
+
+	binds := []string{
+		"/var/run/docker.sock:/var/run/docker.sock",
+		"/:/host:z",
+	}
```

Comment:
For all of these static strings (shown throughout the PR), can we make them all constants or vars declared on the top of the file when possible?

e.g. even some of the bash commands should be declared as constants at the top IMO.

If this creates too much stuff at the top of the file, maybe this file should also be split into multiple files for each one of the calls (e.g. firewall.go, paths.go, etc.)